### ISSUE TITLE
installing FILM

### ISSUE THREAD

**Issue by vis223** [Animator-Anon/animator_extension#8]
Is there any way to make installing FILM a little easier? 
Google's site is so complicated after 2 days I've given up.
I got frame_interpoltion installed but I can find no .bat file anywhere.  I'm new to coding. 

I've got really high hopes for this extension but am hoping you add clearer documentation. Somehow it became even more confusing after you added more info :-) 

If anyone reading this knows how to get FILM installed or resources explaining this style of keyframe prompting it would be much appreciated

**Comment by Animator-Anon** [comment]
Installing FILM is a bit of a pain, I ended up doing it in another python virtual environment by accident, different from the one WebUI is running in. WebUI seems to create it's own on the fly anyway. Because of that, I had to make my batch file change to that VENV, run it and then exit.
Requiring a batch file is just a way for the extension to be able to consistently execute it. How you run it in your setup might be different than mine.
Say you are using windows, If you open a clean command line, what ever commands you need to type in to execute FILM should be in the batch file,. just replace the picture location with %1 and the number of frames to generate with %2. Here is my example:
```
call %USERPROFILE%\miniconda3\condabin\conda.bat activate ldm
python -m eval.interpolator_cli --pattern "%1" --model_path pretrained_models\film_net\Style\saved_model --times_to_interpolate %2
call %USERPROFILE%\miniconda3\condabin\conda.bat deactivate
```

FILM interpolation didn't end up being that useful. It only seems to interpolate from frame to frame, the motion doesn't seem to flow over multiple frames. So the results are very janky, but smooth. Maybe that's my fault, i don't know. I haven't spent a great deal of time on it.

**Comment by vis223** [comment]
thanks for the detailed reply 
Installing was indeed a royal pain. But I got it working and was able to edit your code to suit Anaconda which felt like quite an achievement. So thanks for that. 
FILM is such a weird effect, it's almost creepy how smooth the results so it will take some experimentation.  Your extension creates some very smooth video without it so could be overkill but it's become the industry standard so it's definitely worth having installed.

Having teased you previously about your instructions I'm happy to say I took another look and once the penny dropped it became deceptively quite simple really. The only part I'm not getting is the prompt interpolation. 
Not sure how I missed them before but your instructions are pretty clear and complete now,  I've just noticed all kinds of goodies like png info and source files which I didn't see before or didn't understand. So now I'm excited to roll up my sleeves.

Well done on a truly outstanding extension. Can't wait to see where you take it. I hope others contribute to the parts where you've struggled, like the preview window, It's no biggie to use the image browser in the meantime but I suspect this will become very popular very soon once word gets out. I've got a few ideas for feature requests but I don't want to write a novel just now and already I've already taken enough of your time.

How about adding a discussion thread so users can share tips etc without it being an 'issue'?

Keep up the good work, fight the good fight. Chat GPT had no part in this. All in my own words :-)

**Comment by Animator-Anon** [comment]
I may have stealth edited the instructions over the last few days. That could be the reason!

Prompt interpolation is something built into WebUI, or into the models. I am not exactly sure. Some diffusers don't seem to support, like the ancestral ones.
Basically, you can enter two prompts and a weight for each, and it will generate an in-between image.
So lets say I want to generate a 50% cat and 50% dog, I could type into the webUI txt2img tab:
`cat:0.5 AND dog:0.5`
I can vary the strength from 0 to 1 over time to get a morph between cat and dog also. Ideally, you would have a fixed seed when doing this and NOT be in loopback mode, since it is generating the image purely from the prompts.

This is different to the seed travelling, which is attempting to change the seed from one value to the next in a way that doesn't completely destroy the image. As before, it generates in-between images. You can also do this in the txt2img page by using the subseed and subseed strength values, hidden under the extras checkbox. The seed will be the first seed, the subseed will be the second or target seed. The strength value will be percentage distance from one seed to the next. You can also animate this, and it gives interesting results, as small parts of the image change one after the other. Also as before, probably best to do this NOT in loopback mode and with fixed prompts.

In my experience, after exploring all of these, I couldn't really get the system to generate contextual animations. You really are stuck in the realm of creating Peter Gabriel style music videos, or dream sequences. That in itself is quite interesting though, and people more artistic than me have created some amazing things.

Is there a way to add a discussion thing in github? I won't be starting a discord or anything, this is probably the best way to contact me. feel free to leave the issue open.

**Comment by vis223** [comment]
I agree. I used all of these as scripts and it was very disappointing. But you've given us alternate ways of trying them out which is awesome. I'm playing with an initial image and a target source file to see how things morph. I was always frustrated that seeds never remained consistent and also prompts weren't always predictable but having image files as the basic framework is very interesting. It's all trial and error but I'm pleased with the results so far

**Comment by Animator-Anon** [comment]
Things I have tried:

Manually generating a series of pictures with fixed prompts but random seeds, and picking out the seeds I want, sort of like keyframes in an animation. Then using seed travel to move between them.
You can get good results with luck, but it's a heap of work.

Writing out a series of prompts, kind of story boarding the animation, and prompt interpolating between them. the prompt interpolation can be a bit glitchy, and as much work as above. Works best with a dynamic picture, say a zoom transform that keeps it changing. But limited to dream sequences.

You noticed the ability to read .png info, that was a way to speed up this process. Just dump a few pre-generated pictures into a folder, call them up by filename to have the seed and prompts set at that time.

Things I haven't tried, others may have:
Animating the initial noise, or creating a basic scene with polygons that can be moved around, and then use img2img to build a picture from that. I guess you could animate some characters walking around in basic manner, and have the img2img turn this into a proper scene.

Some method of examining the next frame and checking if it is a good match to the previous one. If not, regenerate with new seed. Kind of brute force a series of pictures that in some way flow together. Could be interesting, could be nightmare fuel.


The idea of the script is a toolbox that allows you to experiment with this stuff. Let me know if you have any ideas on new things to try.

**Comment by vis223** [comment]
Wow, that's quite a lot to digest. I'll take it to step by step. 
I'm not sure if anything changed or if it's automatic1111 but results seem to be going downhill for me a bit. I always seem to end up zooming in to a point that then burns in. But was previously getting a lot more variety for longer time frames.
I'm not clear on how to zoom out. If I zoom in using transform 2, can I then use -2 to zoom back? This doesn't seem to work and the doc only mentions <1 which I'm not sure is meant as the syntax. I think more examples of how to use the different keyframe types would be a great help.
For example, I want to try your example of seed traveling but not sure how that would look. Would that be a list of "time_s | seed | new_seed_int" stacked up? Actually, I could probably test that out for an answer quicker than your reply. I've had better luck with seed traveling than with prompt interpolation.

**Comment by Animator-Anon** [comment]
Zooming is just scaling the picture, so 0.5 would be the inverse of 2. It would make the new picture 50% the size over one second, zooming out.

seed travelling would only be usefull if you have a list of seeds for a prompt that you know work, and want to animate between them. Lets say you have a fixed prompt, and generate 20 different images. If you pick three out and want to animate between them in order over 4 seconds, you could do this:
0 | template | positive prompts here | negative prompts here
0 | prompt | more positve prompts | more negative prompts
0 | seed | 123
2 | seed | 456
4 | seed | 789

If seed travelling is turned off, you'll get the first seed and a static image for 2s, then the second etc. If it's turned on, you'll get a series of different images leading from one seed to the next. You can always look at the .csv file dumped into the output folder in excel or libreoffice calc to see what parameters were used on each frame.

UI breaks stuff sometimes. There was a change a few months back to do with colour correction that broke this extension pretty badly. Try roll the UI back, figure out what the change is.

**Comment by Leka1B** [comment]
I wanted to ask this in a sepparate issue, but mr. Vis mentioned some changes to the command that may be relevant here.

I tried to install FILM today and after some hours got a bat file that works with a set folder and number of frames. I basically used your command and changed all the paths. 
But when I change the paths to %1 and %2 and try to use it in the extension it stops and deactivates right when you would normally see generation progress meters. Everything else looks exacly like using it through the command prompt

**Comment by Animator-Anon** [comment]
So if you open up a fresh command line in windows, and run the batch file with some arguments, does that work? 
You could also add a pause command at the end of the batch file, might help keep it open so you can look for errors messages.
I put %1 in "" marks just in case folder names have spaces in them, that could cause issues.

**Comment by Leka1B** [comment]
Yeah, launching FILM through a batch file works great, the problem appears only when running throuh your script. I compared the console outputs and they are the same, the only difference is when I use the script it stops and disable conda at the moment you would usually see progress bars appear

**Comment by Animator-Anon** [comment]
Did you add the pause command to the bat file and did that keep a window open so you can look for errors?
